### ISSUE TITLE
Bug with Oracle en-Vec

### ISSUE THREAD

**Issue by TheLastNarwhal** [Card-Forge/forge#5290]
**Describe the bug**
**Oracle en-Vec:** *{T}: Target opponent chooses any number of creatures they control. During that player's next turn, the chosen creatures attack if able, and other creatures can't attack. At the beginning of that turn's end step, destroy each of the chosen creatures that didn't attack this turn. Activate only during your turn.*

My opponent selected me for this effect and I chose nearly all my creatures, including my commander, The Necrobloom. During my turn I made a teensy miscalculation when using some MLD while having Ashaya on the field and as a result killed all my creatures. I did end up having enough mana left over to be able to cast The Necrobloom again, but then during the end phase Oracle en-Vec killed my commander. I don't think this is correct since *this* Necrobloom wasn't actually one of the chosen creatures.

**Expected behavior**
To only destroy the chosen creatures, not creatures with that name (which is what I assume happened.)

**Desktop (please complete the following information):**
 - Forge Version: forge-gui-desktop-1.6.63-SNAPSHOT-05-23

**Comment by dracontes** [comment]
> @tool4ever can you look into this? It's probably a timestamp problem?

The issue I found delving into it is that the effect created by Oracle en-Vec is referencing the chosen creatures by `ChosenCard` and isn't forgetting the cards as they incidentally move between zones. 
I tried to reference the chosen creatures by `RememberObjects$ ChosenCard`/`Remembered` so I could use `ForgetOnMoved$ Battlefield` to ensure expected behavior. However in this case, the effect isn't set up when I choose none of my creatures for the activated ability. The expectation would be that when I choose none, my creatures can't attack on my next turn. 
The issue with using `ChosenCard` alone is that there doesn't seem to be an easy way to have the effect using it forget a specific chosen card when that card moves elsewhere, unlike what exists for `Remembered`. I tried adding a suitable static trigger to the Oracle effect to no success. 
A bit late to continue delving into it for today, but I feel that the answer may be in effects that separate creatures into piles like [Raging River](https://scryfall.com/card/2ed/169/raging-river).
